### ISSUE TITLE
The execution result of the program differs from that of the native program.

### ISSUE THREAD

**Issue by abc767234318** [wasmerio/wasmer#3033]
### Describe the bug

```
#include <dirent.h>
#include <stdio.h>
#include <errno.h>

int main(int argc, char **argv) {
  DIR *d;

  char * target = ".";
  if (argc == 2) {
      target = argv[1];
  }
  struct dirent *dir;
  d = opendir(target);
  if (d) {
    while ((dir = readdir(d)) != NULL) {
      printf("%s\n", dir->d_name);
    }

    printf("errno: %d\n", errno);
    closedir(d);
  }

  return(0);
}
```
This is a source code first mentioned in wasmtime issues NO.2493.
My project files are as follows:
[wasmtime-2493.zip](https://github.com/wasmerio/wasmer/files/9159635/wasmtime-2493.zip)

```sh
echo "`wasmer -V` | `rustc -V` | `uname -m`"
wasmer 2.3.0 | rustc 1.62.0 (a8314ef7d 2022-06-27) | x86_64
```

### Steps to reproduce
```
wasmer run ls.wasm --dir=./testfolder -- testfolder | wc -l
```

### Expected behavior
When I compile using g++, the program results in the following
```
203
```

### Actual behavior
```
201
```

**Comment by syrusakbary** [comment]
This indeed seems like a bug, this is how wasmtime fixed it:
https://github.com/bytecodealliance/wasmtime/issues/2493

**Comment by ptitSeb** [comment]
Should be fixed in current master, can you check @abc767234318 so the ticket can be closed?

**Comment by ptitSeb** [comment]
Fixed on my side, closing the ticket now.
